### PR TITLE
Add MemoryMap bracket operators

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
Don't merge yet

**Comment by eile** [comment]
Good to go

File: lunchbox/memoryMap.h [hunk 1]
```diff
@@ -129,6 +129,14 @@ class MemoryMap : public boost::noncopyable
     template< class T > T* getAddress()
         { return static_cast< T* >( getAddress( )); }
 
+    /** Access the given element in the map. @version 1.16 */
+    template< class T > T& get( const size_t i )
+        { return getAddress< T >()[i]; }
+
+    /** Access the given element in the map. @version 1.16 */
+    template< class T > const T& get( size_t i ) const
```
Comment (tribal-tec) [inline lunchbox/memoryMap.h]: no const size_t i here?

File: lunchbox/memoryMap.h [hunk 2]
```diff
@@ -129,6 +129,14 @@ class MemoryMap : public boost::noncopyable
     template< class T > T* getAddress()
         { return static_cast< T* >( getAddress( )); }
 
+    /** Access the given element in the map. @version 1.16 */
+    template< class T > T& operator[]( const size_t i )
+        { return getAddress< T >()[i]; }
+
+    /** Access the given element in the map. @version 1.16 */
+    template< class T > const T& operator[]( size_t i ) const
```
Comment (kaabimg) [inline lunchbox/memoryMap.h]: const size_t ?

Comment (kaabimg) [inline lunchbox/memoryMap.h]: const size_t ?

Comment (eile) [inline lunchbox/memoryMap.h]: done

**Pull request merged**